### PR TITLE
feat: added edit profile fragment

### LINKED ISSUE #50: Create EditProfile activity
working on it
parent issue #27

### PR DESCRIPTION
**Pull request by simarsingh24**
fixes #50

### CODE REVIEW

File: app/src/main/res/layout/fragment_edit_profile.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            style="?android:attr/progressBarStyle"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center"
+            android:layout_marginTop="@dimen/padding_large"
+            android:visibility="gone" />
+
+    </LinearLayout>
+
+
```
Comment (dreadpool2) [inline app/src/main/res/layout/fragment_edit_profile.xml]: extra line

Comment (simarsingh24) [reply app/src/main/res/layout/fragment_edit_profile.xml]: resolved

File: app/src/main/res/layout/fragment_edit_profile.xml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (nikit19) [inline app/src/main/res/layout/fragment_edit_profile.xml]: is this required?

**Comment by simarsingh24** [comment]
@nikit19 resolved

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    fun isLoggedIn() = authService.isLoggedIn()
+
+    fun updateUser(firstName: String, lastName: String) {
+        val id: Long
+        if (firstName.isEmpty() || lastName.isEmpty()) {
+            message.value = "Please provide first name and last name!"
+            return
+        }
+        try {
+            id = JWTUtils.getIdentity(authHolder.getAuthorization())
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/auth/EditProfileViewModel.kt]: authHolder.getId does the same thing right ?

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+import org.fossasia.openevent.general.utils.JWTUtils
+import org.json.JSONException
+import timber.log.Timber
+
+class EditProfileViewModel(private val authService: AuthService, private val authHolder: AuthHolder) : ViewModel() {
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/auth/EditProfileViewModel.kt]: I think you can use only the authHolder and get the work done

Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/auth/EditProfileViewModel.kt]: we need auth service as updateUser is in authservice

Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/auth/EditProfileViewModel.kt]: resolved

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt [hunk 5]
```diff
@@ -79,6 +79,11 @@ class ProfileFragment : Fragment() {
 
     override fun onOptionsItemSelected(item: MenuItem?): Boolean {
         when (item?.getItemId()) {
+            R.id.edit_profile->{
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: Space

**Pull request merged**